cdk/kittyhawk/src/certificate.ts

```diff
@@ -10,26 +10,24 @@ export class Certificate extends Construct {
     if (props.ingress) {
       // We want to generate a certificate for each host
       for (const h of props.ingress) {
-        // Regex to compute the apex domain
-        const apex_domain = h.host.match(/[\w-]+\.[\w]+$/g);
-        if (apex_domain != null) {
-          let host_string = apex_domain[0].split('.').join('-');
-          new CertApiObject(this, `certificate-${appname}-${host_string}`, {
-            metadata: {
-              name: host_string,
+        let hostString: string = domainToCertName(h.host, h.isSubdomain);
+        new CertApiObject(this, `certificate-${appname}-${hostString}`, {
+          metadata: {
+            name: hostString,
+          },
+          spec: {
+            secretName: hostString.concat('-tls'),
+            dnsNames: [`${removeSubdomain(h.host, h.isSubdomain)}`, `*.${removeSubdomain(h.host, h.isSubdomain)}`],
```

Comment:
Maybe do something like `const finalDomain = removeSubdomain(h.host, h.isSubdomain)` a few lines above this just to avoid the possibility of a typo in these two calls?